src/app/components/search/search.component.ts

```diff
@@ -8,19 +11,51 @@ import { FormGroup, FormControl } from '@angular/forms';
 })
 export class SearchComponent {
   dateInputType!: 'specific' | 'range';
+  cases: CaseData[] = [];
+  loaded = false;
+  errorType = '';
+
+  constructor(private caseService: CaseService) {}
 
   form = new FormGroup({
-    searchText: new FormControl(),
-    courthouseText: new FormControl(),
-    courtroomText: new FormControl(),
-    defendantNameText: new FormControl(),
-    judgeNameText: new FormControl(),
-    keywordsText: new FormControl(),
+    case_number: new FormControl('', Validators.required),
+    courthouse: new FormControl(),
+    courtroom: new FormControl(),
+    judge_name: new FormControl(),
+    defendant_name: new FormControl(),
+    event_text_contains: new FormControl(),
+    date_from: new FormControl(),
+    date_to: new FormControl(),
   });
 
   // Submit Registration Form
   onSubmit() {
-    alert(JSON.stringify(this.form.value));
+    this.caseService
+      .getCasesAdvanced(
+        this.form.get('case_number')?.value || '',
+        this.form.get('courthouse')?.value || '',
+        this.form.get('courtroom')?.value || '',
+        this.form.get('judge_name')?.value || '',
+        this.form.get('defendant_name')?.value || '',
+        this.form.get('date_from')?.value || '',
+        this.form.get('date_to')?.value || '',
+        this.form.get('event_text_contains')?.value || ''
+      )
+      .subscribe(
```

Comment:
I read this earlier on the "standards" - https://hmcts.github.io/standards/standards/angular.html#subscribe-in-template

Don't change it now, but maybe one for the future.

Reply:
Good spot - will use async to subscribe in template in next API and then we can easily make the necessary changes to this API.